internal/isoeditor/rhcos.go

```diff
@@ -541,3 +546,49 @@ func ParseOffsetInfo(headerBytes []byte) (*OffsetInfo, error) {
 	}
 	return offsetInfo, nil
 }
+
+func EmbedIgnition(inputISOPath, outputISOPath, ignitionConfig string) error {
+	coreosIgnitionHeader := make([]byte, ignitionHeaderSize)
+
+	inputISO, err := os.Open(inputISOPath)
+	if err != nil {
+		return err
+	}
+	defer inputISO.Close()
+
+	if _, err = inputISO.ReadAt(coreosIgnitionHeader, isoSystemAreaSize-ignitionHeaderSize); err != nil {
+		return err
+	}
+	ignitionOffsetInfo, err := GetIgnitionArea(coreosIgnitionHeader)
+	if err != nil {
+		return err
+	}
+
+	cpio, err := IgnitionImageArchive(ignitionConfig)
+	if err != nil {
+		return err
+	}
+
+	if len(cpio) > int(ignitionOffsetInfo.Length) {
+		return errors.Errorf("Compressed Ignition config is too large: %v > %v", len(cpio), int(ignitionOffsetInfo.Length))
+	}
+
+	resultISO, err := os.Create(outputISOPath)
+	if err != nil {
+		return err
+	}
+	defer resultISO.Close()
+
+	if _, err = io.Copy(resultISO, inputISO); err != nil {
```

Comment:
can you use isoutil.Copy instead?

Reply:
Started implementing this...here are the issues I have:
1. I must create a handler. Not a big deal but it is an extra step.
2. I already have to open the file to get read the ignition area. Maybe I'm just being silly but it feels odd to have to open it twice.

wdyt?

Reply:
ok, so not that crucial. I just thought we could reuse the existing logic if not too complicated.

Reply:
I think what I did makes sense, let me know what you think.